internal/consensus/reactor_test.go

```diff
@@ -287,8 +306,14 @@ func waitForBlockWithUpdatedValsAndValidateIt(
 
 		for {
 			msg, err := blocksSubs[j].Next(ctx)
-			if !assert.NoError(t, err) {
-				cancel()
+			switch {
+			case errors.Is(err, context.DeadlineExceeded):
+				return
+			case errors.Is(err, context.Canceled):
+				return
+			case err != nil:
+				t.Fatalf("problem waiting for %d subscription: %v", j, err)
+				cancel() // terminate other workers
```

Comment:
This cancel call is still unreachable, however. I think you could just do it before calling `t.Fatal` safely.